pandas/plotting/_core.py

```diff
@@ -574,6 +575,13 @@ def _get_xticks(self, convert_period=False):
 
     @classmethod
     def _plot(cls, ax, x, y, style=None, is_errorbar=False, **kwds):
+        # GH25587: cast ExtensionArray of pandas (IntegerArray, etc.) to
+        # np.ndarray before plot.
+        if isinstance(x, ExtensionArray):
```

Comment:
Use `is_extension_array_dtype(x.dtype)` and `np.asarray(x)` to convert.

Reply:
Fixed.